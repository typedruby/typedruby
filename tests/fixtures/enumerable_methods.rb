# @typedruby

def test1 => nil
  [1,2,3].each_with_object({}) do |num, hash|
    hash[num] = num
  end
end

def test2 => nil
  [1,2,3,4].group_by { |n| n < 3 }
end
