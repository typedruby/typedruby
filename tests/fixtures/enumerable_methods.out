error: Could not match types:

        @ __ROOT__/tests/fixtures/enumerable_methods.rb:3
      3 |  def test1 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/enumerable_methods.rb:4
      4 |    [1,2,3].each_with_object({}) do |num, hash|
                                      ^^ Hash::[Integer, Integer]

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/enumerable_methods.rb:3
      3 |  def test1 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/enumerable_methods.rb:4
          ---v
      4 | |  [1,2,3].each_with_object({}) do |num, hash|
      5 | |    hash[num] = num
      6 | |  end
          -----^ Hash::[Integer, Integer]

error: Could not match types:

        @ __ROOT__/tests/fixtures/enumerable_methods.rb:9
      9 |  def test2 => nil
                        ^^^ NilClass, with:
        @ (builtin stdlib):__LINE__
    113 |    def group_by[GroupKey]({ |EnumType element| => GroupKey } &) => { GroupKey => [EnumType] }; end
                                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^ Hash::[Boolean, Array::[Integer]]

        - arising from an attempt to match:

        @ __ROOT__/tests/fixtures/enumerable_methods.rb:9
      9 |  def test2 => nil
                        ^^^ NilClass, with:
        @ __ROOT__/tests/fixtures/enumerable_methods.rb:10
     10 |    [1,2,3,4].group_by { |n| n < 3 }
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Hash::[Boolean, Array::[Integer]]
